lib/honeykiq/server_middleware.rb

```diff
@@ -97,6 +97,13 @@ def on_error(event, error)
       )
     end
 
+    def call_extra_fields(job)
+      case method(:extra_fields).arity
+      when 0 then extra_fields
+      else extra_fields(job)
+      end
+    end
```

Comment:
We do this to avoid introducing a breaking change.

Reply:
Good call.